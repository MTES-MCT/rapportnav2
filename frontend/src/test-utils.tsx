import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
