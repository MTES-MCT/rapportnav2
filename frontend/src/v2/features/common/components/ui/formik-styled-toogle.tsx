import { ToggleLabel } from '@common/components/ui/toogle-label'
import { FormikToggle, ToggleProps } from '@mtes-mct/monitor-ui'
import styled from 'styled-components'

export const StyledFormikToggle = styled(({ label, ...props }: ToggleProps) => (
  <div style={{ width: '100%', display: 'flex', flexDirection: 'row', alignItems: 'end' }}>
    <FormikToggle {...props} label="" isLight isLabelHidden readOnly={false} />
    <ToggleLabel>{label}</ToggleLabel>
  </div>
))(({ theme }) => ({
  marginRight: 8
}))
