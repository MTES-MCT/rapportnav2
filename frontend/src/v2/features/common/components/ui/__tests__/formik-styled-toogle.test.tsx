import { Formik } from 'formik'
import { vi } from 'vitest'
import { render } from '../../../../../../test-utils'
import { StyledFormikToggle } from '../formik-styled-toogle'

const handleSubmit = vi.fn()
describe('StyledFormikToggle', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <Formik initialValues={{ toogle: false }} onSubmit={handleSubmit}>
        <StyledFormikToggle name="toogle" label="this is a style formik" />
      </Formik>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
