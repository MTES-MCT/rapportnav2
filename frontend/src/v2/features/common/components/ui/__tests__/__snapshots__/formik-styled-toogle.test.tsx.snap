// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyledFormikToggle > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%; display: flex; flex-direction: row; align-items: end;"
      >
        <div
          class="sc-fYmhhH jaaEau Element-Field Field-Toggle sc-lehtBJ ezRwHe"
        >
          <label
            class="sc-gahYZc eKorlU Element-Label"
            for="toogle"
          />
          <label
            class="sc-iyBeIh bNdnHy rs-toggle"
          >
            <input
              aria-checked="false"
              aria-disabled="false"
              aria-label="Fermer"
              class="rs-toggle-input"
              name="toogle"
              role="switch"
              type="checkbox"
            />
            <span
              class="rs-toggle-presentation"
            >
              <span
                class="rs-toggle-inner"
              />
            </span>
          </label>
        </div>
        <p
          class="sc-evkzZa sc-enPhjR fjwyIl esqQrj sc-kbeVdv eyllFw"
          font-size="13"
        >
          this is a style formik
        </p>
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%; display: flex; flex-direction: row; align-items: end;"
    >
      <div
        class="sc-fYmhhH jaaEau Element-Field Field-Toggle sc-lehtBJ ezRwHe"
      >
        <label
          class="sc-gahYZc eKorlU Element-Label"
          for="toogle"
        />
        <label
          class="sc-iyBeIh bNdnHy rs-toggle"
        >
          <input
            aria-checked="false"
            aria-disabled="false"
            aria-label="Fermer"
            class="rs-toggle-input"
            name="toogle"
            role="switch"
            type="checkbox"
          />
          <span
            class="rs-toggle-presentation"
          >
            <span
              class="rs-toggle-inner"
            />
          </span>
        </label>
      </div>
      <p
        class="sc-evkzZa sc-enPhjR fjwyIl esqQrj sc-kbeVdv eyllFw"
        font-size="13"
      >
        this is a style formik
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
